Public/Src/Tools/Execution.Analyzer/Analyzers.Core/XLGPlusPlus/XLGppDataStore.cs

```diff
@@ -0,0 +1,111 @@
+
+        public XLGppDataStore()
+        {
+
+        }
+
+        /// <summary>
+        /// Open the datastore and populate the KeyValueStoreAccessor for the XLG++ DB
+        /// </summary>
+        /// <returns>Boolean if datastore was opened successfully</returns>
+        public bool OpenDatastore(string storeDirectory,
+            bool defaultColumnKeyTracked = false,
+            IEnumerable<string> additionalColumns = null,
+            IEnumerable<string> additionalKeyTrackedColumns = null,
+            Action<Failure> failureHandler = null,
+            bool openReadOnly = false,
+            bool dropMismatchingColumns = false,
+            bool onFailureDeleteExistingStoreAndRetry = false)
+        {
+
+            var accessor = KeyValueStoreAccessor.Open(storeDirectory, 
+                defaultColumnKeyTracked, 
+                additionalColumns, 
+                additionalKeyTrackedColumns, 
+                failureHandler, 
+                openReadOnly, 
+                dropMismatchingColumns, 
+                onFailureDeleteExistingStoreAndRetry);
+
+            if (accessor.Succeeded)
+            {
+                Accessor = accessor.Result;
+                return true;
+            }
+            else
+            {
+                return false;
+            }
+        }
+
+        /// <summary>
+        /// Gets all the events of a certain type from the DB
+        /// </summary>
+        /// <returns>List of event objects recovered from DB </returns>
+        public List<string> GetEventsByType(int eventTypeID)
+        {
+            Contract.Assert(Accessor != null, "XLGppStore must be initialized via OpenDatastore first");
+
+            var storedEvents = new List<string>();
+            var eventQuery = new EventTypeQuery
+            {
+                EventTypeID = eventTypeID,
+            };
+            Analysis.IgnoreResult(
+                Accessor.Use(database =>
+                {
+                    foreach (var kvp in database.PrefixSearch(eventQuery.ToByteArray()))
+                    {
+                        Console.WriteLine(BXLInvocationEvent.Parser.ParseFrom(kvp.Value));
+                        storedEvents.Add(BXLInvocationEvent.Parser.ParseFrom(kvp.Value).ToString());
+                    }
+                })
+            );
+            
+            return storedEvents;
+        }
+
+        /// <summary>
+        /// Method to test if the appropriate things have been stored in DB.
+        /// NOTE: For internal testing/Debugging only!
+        /// </summary>
+        /// <returns>Stored data in string format</returns>
+        public string GetStoredData()
+        {
+            Contract.Assert(Accessor != null, "XLGppStore must be initialized via OpenDatastore first");
+
+            string value = null;
+            Analysis.IgnoreResult(
+                Accessor.Use(database =>
+                {
+                    database.TryGetValue("foo", out value);

```

Comment:
>foo [](start = 42, length = 3)

I don't like this `foo` and `b` --- if you need to pass some 'dummy' parameters they should be passed from the unit test, i.e., this class should not have 'test' methods.

Reply:
Oh this was just my playground kind of thing for debugging that some things were inserted into the db and what not. I was going to leave this in here until everything was fully developed before removing this (or maybe moving it into some other utility to go through the DB). 

---
In reply to: [306051085](https://github.com/microsoft/BuildXL/pull/636#discussion_r306051085) [](ancestors = 306051085)

Reply:
Lets move this into a true unit test in the next PR

---
In reply to: [306066464](https://github.com/microsoft/BuildXL/pull/636#discussion_r306066464) [](ancestors = 306066464,306051085)